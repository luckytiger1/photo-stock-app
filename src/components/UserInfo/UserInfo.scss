.user-info__text {
  display: flex;
  flex-direction: column;
}
.image-page__actions {
  .user-info {
    display: flex;
    color: white;
    font-family: 'SF UI Display Light', Helvetica, sans-serif;
    height: 100%;

    .user-name {
      font-size: 20px;
      font-family: 'SF UI Display Light', Helvetica, sans-serif;
    }

    img {
      border: 1px solid white;
      border-radius: 5px;
      margin: 5px 10px 0px;
      width: 45px;
      height: 45px;
      vertical-align: middle;
    }
  }
}
