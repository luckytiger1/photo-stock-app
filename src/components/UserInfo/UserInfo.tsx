import React from 'react';
import './UserInfo.scss';

const UserInfo = ({ image }: any) => {
  return (
    <div className="user-info">
      <img src={image.user.profile_image.small} alt="user-pic" />
      <div className="user-info__text">
        <span className="user-name">{image.user.name}</span>
        <span className="user-instagram">
          @
          {image.user.instagram_username
            ? image.user.instagram_username
            : image.user.username}
        </span>
      </div>
    </div>
  );
};

export default UserInfo;
