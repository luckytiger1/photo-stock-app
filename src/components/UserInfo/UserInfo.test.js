import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import UserInfo from './UserInfo';

let mockImage;

beforeEach(() => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
  mockImage = {
    user: {
      profile_image: {
        small: 'image.jpg',
      },
      name: 'Dan',
      instagram_username: 'DanS',
      username: 'DanDan',
    },
  };
});

it('<UserInfo /> renders correctly', () => {
  expect(shallow(<UserInfo image={mockImage} />)).toMatchSnapshot();
});
