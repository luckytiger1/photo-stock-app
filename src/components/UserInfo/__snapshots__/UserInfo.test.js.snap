// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> renders correctly 1`] = `
<div
  className="user-info"
>
  <img
    alt="user-pic"
    src="image.jpg"
  />
  <div
    className="user-info__text"
  >
    <span
      className="user-name"
    >
      Dan
    </span>
    <span
      className="user-instagram"
    >
      @
      DanS
    </span>
  </div>
</div>
`;
